From cf7a1d6da70ea1c59145deb23429f6c3a49ba51e Mon Sep 17 00:00:00 2001
From: Joyce Ooi <joyce.ooi@intel.com>
Date: Wed, 7 Oct 2020 00:33:47 +0800
Subject: [PATCH 138/151] net: eth: altera: fix build errors in QSE driver

commit  7fb6d926df5dd64d480519596f3aea9e97f6552a from
https://github.com/altera-opensource/linux-socfpga.git
branch is socfpga-5.4.64-lts

This patch fixes the functions that cause build errors as this driver
was backported from newer kernel version.

Signed-off-by: Joyce Ooi <joyce.ooi@intel.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
Integrated-by: Jun Zhang <jun.zhang@windriver.com>
---
 drivers/net/ethernet/altera/intel_fpga_qse_ll_main.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/drivers/net/ethernet/altera/intel_fpga_qse_ll_main.c b/drivers/net/ethernet/altera/intel_fpga_qse_ll_main.c
index 8d14b46a1091..e465a8918449 100644
--- a/drivers/net/ethernet/altera/intel_fpga_qse_ll_main.c
+++ b/drivers/net/ethernet/altera/intel_fpga_qse_ll_main.c
@@ -1290,7 +1290,7 @@ static void intel_fpga_qse_validate(struct phylink_config *config,
 		   __ETHTOOL_LINK_MODE_MASK_NBITS);
 }
 
-static void intel_fpga_qse_mac_pcs_get_state(struct phylink_config *config,
+static int intel_fpga_qse_mac_link_state(struct phylink_config *config,
 					    struct phylink_link_state *state)
 {
 	struct intel_fpga_qse_private *priv =
@@ -1306,6 +1306,7 @@ static void intel_fpga_qse_mac_pcs_get_state(struct phylink_config *config,
 
 	state->link = 1;
 
+	return 0;
 }
 
 static void intel_fpga_qse_mac_an_restart(struct phylink_config *config)
@@ -1371,7 +1372,7 @@ static void intel_fpga_qse_mac_link_up(struct phylink_config *config,
 
 static const struct phylink_mac_ops intel_fpga_qse_phylink_ops = {
 	.validate = intel_fpga_qse_validate,
-	.mac_pcs_get_state = intel_fpga_qse_mac_pcs_get_state,
+	.mac_link_state = intel_fpga_qse_mac_link_state,
 	.mac_an_restart = intel_fpga_qse_mac_an_restart,
 	.mac_config = intel_fpga_qse_mac_config,
 	.mac_link_down = intel_fpga_qse_mac_link_down,
@@ -1611,9 +1612,10 @@ static int intel_fpga_qse_ll_probe(struct platform_device *pdev)
 	spin_lock_init(&priv->ptp_priv.tod_lock);
 
 	/* check if phy-mode is present */
-	ret = of_get_phy_mode(np, &priv->phy_iface);
-	if (ret) {
+	priv->phy_iface = of_get_phy_mode(np);
+	if (priv->phy_iface < 0) {
 		dev_err(&pdev->dev, "incorrect phy-mode\n");
+		ret = -EINVAL;
 		goto err_free_netdev;
 	}
 
-- 
2.26.1

